src/main/java/net/mcreator/element/GeneratableElement.java

```diff
@@ -90,8 +92,17 @@ public boolean wasConversionApplied() {
 	public static class GSONAdapter
 			implements JsonSerializer<GeneratableElement>, JsonDeserializer<GeneratableElement> {
 
-		protected static final Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().setLenient()
-				.create();
+		private static final Gson gson;
+
+		static {
+			GsonBuilder gsonBuilder = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().setLenient();
+
+			for (Map.Entry<Class<? extends RetvalProcedure<?>>, JsonDeserializer<? extends RetvalProcedure<?>>> entry : RetvalProcedure.GSON_ADAPTERS.entrySet()) {
+				gsonBuilder.registerTypeAdapter(entry.getKey(), entry.getValue());
+			}

```

Comment:
Maybe this will look better if inlined?
```suggestion
			RetvalProcedure.GSON_ADAPTERS.forEach(gsonBuilder::registerTypeAdapter);
```